nssp/delphi_nssp/run.py

```diff
@@ -0,0 +1,168 @@
+    """Short util to add expected columns not found in the dataset."""
+    if col_names is None:
+        col_names = [
+            "se", 
+            "sample_size",
+            "missing_val",
+            "missing_se",
+            "missing_sample_size"]
+
+    for col_name in col_names:
+        df[col_name] = np.nan
+    df = add_default_nancodes(df)
+    return df
+
+
+def logging(start_time, run_stats, logger):
+    """Boilerplate making logs."""
+    elapsed_time_in_seconds = round(time.time() - start_time, 2)
+    min_max_date = run_stats and min(s[0] for s in run_stats)
+    csv_export_count = sum(s[-1] for s in run_stats)
+    max_lag_in_days = min_max_date and (datetime.now() - min_max_date).days
+    formatted_min_max_date = min_max_date and min_max_date.strftime("%Y-%m-%d")
+    logger.info(
+        "Completed indicator run",
+        elapsed_time_in_seconds=elapsed_time_in_seconds,
+        csv_export_count=csv_export_count,
+        max_lag_in_days=max_lag_in_days,
+        oldest_final_export_date=formatted_min_max_date,
+    )
+
+
+def run_module(params):
+    """
+    Run the indicator.
+
+    Arguments
+    --------
+    params:  Dict[str, Any]
+        Nested dictionary of parameters.
+    """
+    start_time = time.time()
+    logger = get_structured_logger(
+        __name__,
+        filename=params["common"].get("log_filename"),
+        log_exceptions=params["common"].get("log_exceptions", True),
+    )
+    export_dir = params["common"]["export_dir"]
+    socrata_token = params["indicator"]["socrata_token"]
+    # if "archive" in params:
+    #     daily_arch_diff = S3ArchiveDiffer(
+    #         params["archive"]["cache_dir"],
+    #         export_dir,
+    #         params["archive"]["bucket_name"],
+    #         "nchs_mortality",
+    #         params["archive"]["aws_credentials"],
+    #     )
+    #     daily_arch_diff.update_cache()
+
+    run_stats = []
+    ## build the base version of the signal at the most detailed geo level you can get.
+    ## compute stuff here or farm out to another function or file
+    df_pull = pull_nssp_data(socrata_token)
+    ## aggregate
+    geo_mapper = GeoMapper()
+    for signal_i, signal in enumerate(SIGNALS):
+        for geo in GEOS:
+            df = df_pull.copy()
+            df["val"] = df[signal]
+            logger.info("Generating signal and exporting to CSV", metric=signal)
+            if geo == "nation":
+                df = df[df["geography"] == "United States"]
+                df["geo_id"] = "us"
+            elif geo == "state":
+                df = df[(df['county'] == "All") & (df["geography"] != "United States")]
+                df["geo_id"] = df["geography"].apply(lambda x: us.states.lookup(x).abbr.lower() if us.states.lookup(x) else 'dc')
+            elif geo == "hrr" or geo == "msa":
+                df = df[['fips', 'val', 'timestamp']]
+                df = geo_mapper.add_population_column(df, geocode_type="fips", geocode_col="fips")
```

Comment:
So I found [this](https://www.cdc.gov/nchs/data/data_acces_files/County-Geography.pdf) which discusses several of these:
- 02201  former Prince of Wales-Outer Ketchikan Census Area (absorbed in 2006Alaska)
- 02232 Skagway-Hoonah-Angoon Census Area (split)
- 02270 Wade-Hampton Census Area, AK (assigned a new code 02158)
- 02280 Wrangell-Petersburg Census Area was split to create part of Wrangell City and Borough
- 51515 In 2013, Bedford City, an independent city, merged with Bedford county (FIPS code=51019)
Most of these are old fips that were kept around for some reason.
missing:
- 02261

I'm going to take a look to see if any of these are actually non-`NA` in the data, but thanks for the catch Nat.

W/out the raw data, I'm not sure how we handle not having populations for the state FIPS (e.g. `50000`; I guess overall minus the actually present counties?

Reply:
Oh, correction on the unassigned state FIPS. We can't actually do anything with them because HRR and MSA are made up of counties or zipcodes. If we were aggregating FIPS -> state, we could use David's suggestion, but NSSP actually reports state values directly.

Reply:
> if any of these are actually non-NA in the data

Good point, I didn't look at that.

Reply:
TLDR: Of the non-state ones, `['02201', '02232', '02261', '02270', '02280', '51515']`, only `51515` is non-`NaN`, and it's an exact copy of `51019`.
Code for `51515`:
```
bedford = df[(df.fips =='51515') | (df.fips == '51019')].sort_values("week_end")
relevant_columns = ["percent_visits_combined", "percent_visits_covid", "percent_visits_influenza", "percent_visits_rsv", "percent_visits_smoothed", "percent_visits_smoothed_covid", "percent_visits_smoothed_1", "percent_visits_smoothed_rsv"]
bedford[bedford.fips == '51019']
bedford.transpose()
(bedford[bedford.fips == '51019'][relevant_columns].reset_index(drop = True) == bedford[bedford.fips == '51515'][relevant_columns].reset_index(drop = True)).all()
```
Example code for the alaska ones (`022**`):
```
not_present_fips = "2280"
petersburg = df[(df.fips ==not_present_fips) | (df.fips == present_fips)].sort_values("week_end")
relevant_columns = ["percent_visits_combined", "percent_visits_covid", "percent_visits_influenza", "percent_visits_rsv", "percent_visits_smoothed", "percent_visits_smoothed_covid", "percent_visits_smoothed_1", "percent_visits_smoothed_rsv"]
petersburg[relevant_columns].isnull().all()

```

Reply:
Great, then we don't need to do anything here.